### PR TITLE
Make querynode LoadSegments interface idempotent

### PR DESCRIPTION
**Pull request by letian-jiang**
Related to #17098

This PR makes the following changes:
* separate LoadSegmentsTask into two phases: PreExecute and Execute
* filters out segments that are already loaded in PreExecute phase

Signed-off-by: Letian Jiang <letian.jiang@zilliz.com>

### CODE REVIEW

File: internal/querynode/task.go [hunk 1]
```diff
@@ -656,13 +650,29 @@ func (l *loadSegmentsTask) Execute(ctx context.Context) error {
...
 	}
 
-	err = l.node.loader.loadSegment(l.req, segmentTypeSealed)
+	// filter segments that are already loaded in this querynode
+	var filteredInfos []*queryPb.SegmentLoadInfo
+	for _, info := range l.req.Infos {
+		if !l.node.historical.replica.hasSegment(info.SegmentID) {
+			filteredInfos = append(filteredInfos, info)
+		} else {
+			log.Debug("ignore segment that is already loaded", zap.Int64("segmentID", info.SegmentID))
```
Comment (congqixia) [inline internal/querynode/task.go:659]: NIT: add more information about segment such as collectionID and partitionID for debugging.

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/querynode/task.go (hunk 1 above)
Comment (letian-jiang) [reply internal/querynode/task.go:659]: Will fix in the next PR

**Comment by xiaofan-luan** [comment]
I really appreciate the idea to make everything idempotent so developer could retry forever when they recover from failure without handling the case happening~

**Pull request merged**